Bugfix: Escape spaces in directory path.

BrianHenryIE opened the pull request:
The `cd` command was reaching the wrong directory when the directory path contained a space.

BrianHenryIE committed:
Escape spaces in target directory path.

The script was not working when run in folder where a parent folder name contained a space.

BrianHenryIE committed:
Update ScriptHandler.php

BrianHenryIE commented:
Regex can be verified at: https://www.phpliveregex.com/p/uaf (click on _preg_replace_ tab).

kporras07 commented:
Hi @BrianHenryIE thanks for the contribution :D 
Could you fix the coding standard issue in the Travis test? Please

BrianHenryIE committed:
phpcs

BrianHenryIE committed:
Unit test for escaping spaces in paths

BrianHenryIE committed:
Bugfix: data loss when target contains trailing slash

I pretty comprehensively tested this with a mix of leading a trailing slashes and found that when the symlink already exists, if the target in the composer json dictionary has a trailing slash, it gets deleted (MacOS anyway).

I can't think of any instance (file or folder) where the destination would need a trailing slash, so I'm removing it.

BrianHenryIE commented:
To be clear: the target being deleted only happens when the symlink exists. The first time things run, it's fine. I tested 24 permutations of leading and trailing slashes.

BrianHenryIE committed:
Move the bugfix earlier in the function.

The test wasn't wrong, it just wasn't very good.

BrianHenryIE commented:
Earlier code was placed too late in the function so has been moved earlier.

kporras07 commented:
Hey! This looks pretty awesome! I'll take a look next week :D Thank you very much 👌

kporras07 commented:
I just tested it and it works :D Thanks again for your contribution.
